### DISCUSSION TITLE
We're using an old version(2.3).  When did the code get converted to classes?

### DISCUSSION THREAD

**Discussion by scottnc27603** [konvajs/konva#1205] [General]
I tried to use the new version and I got a few errors.  For one BaseLayer was no longer defined.  I'm just wondering if I can pull a version tag and build a new version without updating all of our code.

**Comment by lavrton** [comment]
Well, 2.3 is an old one. Current version is 8. There were not many breaking changing along the way. So updating to the last version shouldn't be very hard. You can try fast force update with just using the last version, and then checking change log for any issues.
Or you can try upgrade version one by one and look for deprecation warnings.

`BaseLayer` was never public and documented. I am not sure why did you use it.
You can get any old version from the repository and build it by yourself.

**Comment by scottnc27603** [comment]
Thanks!  I'll try going one by one.  The main implementation was over 5 years ago, so it's some legacy code for sure.
